src/modules/fw_att_control/FixedwingAttitudeControl.cpp

```diff
@@ -638,6 +673,32 @@ void FixedwingAttitudeControl::run()
 				control_input.groundspeed = groundspeed;
 				control_input.groundspeed_scaler = groundspeed_scaler;
 
+				/* bi-linear interpolation over airspeed for actuator trims */
+				float trim_roll = _parameters.trim_roll;
+				float trim_pitch = _parameters.trim_pitch;
+				float trim_yaw = _parameters.trim_yaw;
+
+				if (airspeed < _parameters.airspeed_trim) {
+					trim_roll = math::gradual(airspeed, _parameters.airspeed_min, _parameters.airspeed_trim,
+								  _parameters.trim_roll_vmin, _parameters.trim_roll);
+					trim_pitch = math::gradual(airspeed, _parameters.airspeed_min, _parameters.airspeed_trim,
+								   _parameters.trim_pitch_vmin, _parameters.trim_pitch);
+					trim_yaw = math::gradual(airspeed, _parameters.airspeed_min, _parameters.airspeed_trim,
+								 _parameters.trim_yaw_vmin, _parameters.trim_yaw);
+
+				} else {
+					trim_roll = math::gradual(airspeed, _parameters.airspeed_trim, _parameters.airspeed_max,
+								  _parameters.trim_roll, _parameters.trim_roll_vmax);
+					trim_pitch = math::gradual(airspeed, _parameters.airspeed_trim, _parameters.airspeed_max,
+								   _parameters.trim_pitch, _parameters.trim_pitch_vmax);
+					trim_yaw = math::gradual(airspeed, _parameters.airspeed_trim, _parameters.airspeed_max,
+								 _parameters.trim_yaw, _parameters.trim_yaw_vmax);
+				}
+
+				/* add trim increment if flaps are deployed  */
+				trim_roll += _flaps_applied * _parameters.dtrim_roll_flaps;
```

Comment:
Do you really need a trim on roll since the flaps should be quite symmetric?

Reply:
Ideally - yes the flaps should be a symmetric effect, but can be (depending on how large the flap setting is) that there is some non-symmetric aerodynamic behavior from the platform geometry (or potentially just simply the servos not going to exactly the same place). I'm not opposed to removing it though - as normally, as you say, it should be fine only to trim out the elevator. Thoughts?

Reply:
I would say that we could leave it like that; even if I think that directly trimming the servos of the flaps would be cleaner, it might be easier do trim them this way.
If we see that it's not useful, we can still remove it later.